src/mod/endpoints/mod_sofia/mod_sofia.h

```diff
@@ -758,6 +759,7 @@ struct sofia_profile {
 	uint32_t event_timeout;
 	int watchdog_enabled;
 	switch_mutex_t *gw_mutex;
+	switch_mutex_t *gw_deleting_mutex;
```

Comment:
Do we need this? I think we don't have to introduce another mutex for this.

Reply:
sofia_reg_add_gateway and sofia_reg_check_gateway can happen in different threads. Check this: https://github.com/team-telnyx/freeswitch/pull/63/files#r597707629